### PR TITLE
Support for publishing multiple specs from a directory

### PR DESCRIPTION
**Pull request by bajalnyt**
Adds an [optional] parameter `specs_dir` to the step. If provided, the code scans through all files in the given directory and pushes them one by one. 

We have a monorepo that has several yamls, and it is tedious to add a step each time a new spec is created. [See here](https://github.com/nytimes/spg-specs/blob/main/.drone.yml#L41)

Need help verifying, is it possible to push the image with a tag, so that I can try it out from our repo?

Closes https://github.com/nytimes/drone-openapi/pull/5.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,8 +11,8 @@ This plugin supports Drone 0.4 and 0.6+ (0.5 is deprecated).
 The example below is for secrets in the Drone 1.0+ format, where the GCP Service Account json must be passed to the `GOOGLE_CREDENTIALS` parameter in .drone.yml as an environment variable.
 
 
-### Basic example config to publish the swagger.yaml spec file under the kids team:
-
+### Basic example config to publish the swagger.yaml spec file:
+```
```
Comment (timothyeburke) [inline README.md:15]: If you replace with 
```
```YAML
```
Github will code highlight the code block, improving readability.

File: README.md [hunk 2]
```diff
@@ -27,3 +27,23 @@ The example below is for secrets in the Drone 1.0+ format, where the GCP Service
       - push
       branch:
       - main
+```
+
+To publish all yamls in a given directory:
+
+```
```
Comment (timothyeburke) [inline README.md:34]: Same here.

File: main.go [hunk 3]
```diff
@@ -74,11 +77,33 @@ func wrapMain() error {
 		return err
 	}
 
+	if len(vargs.Directory) > 0 { // Process multiple yaml files in a given directory
```
Comment (tonglil) [inline main.go:80]:
```suggestion
	if len(vargs.Directory) > 0 {
		fmt.Printf("processing multiple specs in directory: %s\n", vargs.Directory)
```

File: main.go [hunk 4]
```diff
@@ -74,11 +77,33 @@ func wrapMain() error {
...
+
+		for _, file := range files {
+			err = publishSingleSpec(vargs, file.Name())
+			if err != nil {
+				return err
+			}
+		}
+
+	} else {
+		err = publishSingleSpec(vargs, vargs.Spec)
```
Comment (tonglil) [inline main.go:94]:
```suggestion
		fmt.Printf("processing spec file: %s\n", vargs.Spec)
		err = publishSingleSpec(vargs, vargs.Spec)
```

**Review by tonglil — CHANGES_REQUESTED** [review]
Thanks for submitting!

While I don't particularly like overwriting `vargs.Spec` for each file, it's acceptable.

Other than the comments, can you add some usage/docs in the README?

We probably need to setup some CI for this as well.

**Comment by bajalnyt** [comment]
@tonglil , thanks a lot for the review!

I have made those changes, and added a unit test/integration test for the code in main.go. The integration test won't run without the key/credential passed in, but it can be used to verify that the specs are picked up correctly.

Happy to help build out a CI for this repo, if I you can activate this repo in drone (I don't have the needed access)

**Comment by bajalnyt** [comment]
@tonglil : appreciate a re-review when you get a chance. Also happy to QA this in one of our repos, if this plugin can be pushed with a specific tag before an official release. Thanks!

**Review by tonglil — COMMENTED** [review]
Thanks for all the hard work @bajalnyt.

I'm creating the CI now. Can you take a look at and rebase on-top of https://github.com/nytimes/drone-openapi/pull/9?

Thanks!

File: .gitignore [hunk 5]
```diff
@@ -1 +1,19 @@
...
+# OS generated files #
+######################
+.DS_Store
+.DS_Store?
+._*
+.Spotlight-V100
+.Trashes
+ehthumbs.db
+Thumbs.db
+
```
[the remark this answers is not in the document]
Comment (bajalnyt) [reply .gitignore]: Done. I've rebased the `updates` branch here

File: .gitignore [hunk 6]
```diff
@@ -1 +1,4 @@
 drone-openapi
+coverprofile.out
+testdata/specs/hello_world_spec.json
+testdata/specs/users_spec.json
```
Comment (tonglil) [inline .gitignore:4]: Do you want the json files?

**Review by tonglil — APPROVED** [review]

File: .gitignore (hunk 6 above)
Comment (bajalnyt) [reply .gitignore:4]: No, the json actually gets generated during the test from the ymls which are checked in

**Pull request merged**